src/main.js

```diff
@@ -36,3 +36,20 @@ app.initialize();
 
 // export app to console
 window.app = app;
+
+// When an <img> finishes loading, remove the `loading` class from its parent.
+document.body.addEventListener('load', function(evt){
+  if (evt.target.tagName === 'IMG') {
+    evt.target.parentNode.classList.remove('loading');
+  }
+}, true);
+
+// When an <img> tag errors, replace the `loading` class from its parent with
+// `error` and remove the element from the DOM entirely.
+document.body.addEventListener('error', function(evt){
+  if (evt.target.tagName === 'IMG') {
+    evt.target.parentNode.classList.remove('loading');
+    evt.target.parentNode.classList.add('error');
+    evt.target.parentNode.removeChild(evt.target);
+  }
+}, true);
```

Comment:
It might be cleaner to do this inside the views, rather than top-level in the DOM. Ampersand has a convention around delegating events within a component, so it seems kinda funny to have global code messing with nodes inside components.

If we do this in the views, we'd probably also want to set the img src on the JS side, so that we can be sure the load/error listeners are attached before the image loads (I think this can be tricky if the image is in the browser cache). Maybe something like
```
afterRender() {
  // ... other stuff ...
  this.img = this.el.querySelector('img'); // it's the only img, but add a class if you want
  this.img.onerror = this.onImgFailed;
  this.img.onload = this.onImgLoaded;
  this.img.src = 'https://path/to/img';
}
```
And actually, this seems like a good opportunity to reconsider whether we want to add an ItemView that sits between the base view and the individual item views. We could abstract out this whole favicon loading thing into a function that's called by the super class inside its afterRender:
```
ItemView {
  // ... stuff ...
  afterRender() {
    this.renderMainImage();
  }
  renderMainImage() {
    this.img = this.el.querySelector('img#main'); // I'm not great at naming stuff, but you get the idea
    this.img.onerror = this.onImgFailed;
    this.img.onload = this.onImgLoaded;
    this.img.src = this.mainImageUrl; // subclasses override this in `beforeRender` or in `afterRender`
  }
}

// ActivityItemView subclass of ItemView
ActivityItemView {
  // ... other stuff ...
  afterRender() {
    this.mainImageUrl = this.data.img.url;
    ItemView.prototype.afterRender.apply(this, arguments);
  }
}
```
I dunno, this is just a sketch (I haven't tried it), but might be one way to do it that minimizes code duplication. Feel free to explore whatever approach seems nicest to you (Ampersand supports mixins, and we might want to reuse the image loading code, so that's a possibility that could be worth exploring).